pandas/core/internals/base.py

```diff
@@ -356,3 +361,16 @@ def interleaved_dtype(dtypes: list[DtypeObj]) -> DtypeObj | None:
         return None
 
     return find_common_type(dtypes)
+
+
+def ensure_np_dtype(dtype: DtypeObj) -> np.dtype:
+    # TODO: https://github.com/pandas-dev/pandas/issues/22791
+    # Give EAs some input on what happens here. Sparse needs this.
+    if isinstance(dtype, SparseDtype):
+        dtype = dtype.subtype
+        dtype = cast(np.dtype, dtype)
+    elif isinstance(dtype, ExtensionDtype):
```

Comment:
Doesn't need to handle `NumpyEADtype`?

Reply:
shouldnt be needed.  if a dtype is passed it should be a numpy dtype.  if a dtype isnt passed it will be found with interleave_dtype which should be safe since we don't have NumpyEA here